### PR TITLE
br-ext: optee_client_ext: add config switch to disable RPMB emulation

### PR DESCRIPTION
**Pull request by etienne-lms**
New BR2_PACKAGE_OPTEE_CLIENT_EXT_HW_RPMB boolean configuration switch
on optee_client_ext package to set RPMB_EMU=OFF for CMake build of
optee_client repository.

Signed-off-by: Etienne Carriere <etienne.carriere@linaro.org>

### CODE REVIEW

File: br-ext/package/optee_client_ext/Config.in [hunk 1]
```diff
@@ -8,3 +8,9 @@ config BR2_PACKAGE_OPTEE_CLIENT_EXT_SITE
 	default ""
 	help
 	  The path to the source directory of OP-TEE client
+
+config BR2_PACKAGE_OPTEE_CLIENT_EXT_HW_RPMB
+	bool "Enable real RPMB device"
+	help
+	  Enable real RPMB device support by disabling the RPMB emulation in
+	  tee-supplicant.
```
Comment (jforissier) [inline br-ext/package/optee_client_ext/Config.in:16]: How about `BR2_PACKAGE_OPTEE_CLIENT_EXT_RPMB_EMU` default `y`?

Comment (etienne-lms) [reply br-ext/package/optee_client_ext/Config.in:16]: +1. Would be more consistent. I'll update. thanks.

**Comment by etienne-lms** [comment]
updated and added an extra commit
"br-ext: optee_client_ext: explicitly set CFG_TEE_BENCHMARK"
for consistency in package configuration.

**Comment by jforissier** [comment]
`Reviewed-by: Jerome Forissier <jerome@forissier.org>` (with the commit description of the first commit properly updated of course)

**Comment by etienne-lms** [comment]
Fixup commit squashed, 1st commit log update and R-b tag applied to both changes.

File: br-ext/package/optee_client_ext/Config.in [hunk 2]
```diff
@@ -8,3 +8,9 @@ config BR2_PACKAGE_OPTEE_CLIENT_EXT_SITE
 	default ""
 	help
 	  The path to the source directory of OP-TEE client
+
+config BR2_PACKAGE_OPTEE_CLIENT_EXT_RPMB_EMU
+	bool "Enable real RPMB device"
```
[the remark this answers is not in the document]
Comment (etienne-lms) [reply br-ext/package/optee_client_ext/Config.in:13]: right! thanks.

File: br-ext/package/optee_client_ext/Config.in [hunk 3]
```diff
@@ -10,7 +10,7 @@ config BR2_PACKAGE_OPTEE_CLIENT_EXT_SITE
 	  The path to the source directory of OP-TEE client
 
 config BR2_PACKAGE_OPTEE_CLIENT_EXT_RPMB_EMU
-	bool "Enable real RPMB device"
+	bool "enable RPMB emulation"
```
Comment (jforissier) [inline br-ext/package/optee_client_ext/Config.in:13]: Upper case "Enable" ;)

**Review by jforissier — COMMENTED** [review]
You may squash the fixups and I will merge this.

File: br-ext/package/optee_client_ext/Config.in (hunk 3 above)
Comment (etienne-lms) [reply br-ext/package/optee_client_ext/Config.in:13]: I asked that myself before updating this P-R. Looking into buildroot source tree, the descriptions tend to use lower case 1st char in the config switch short description.
You prefer an upper case char here?

Comment (jforissier) [reply br-ext/package/optee_client_ext/Config.in:13]: Not so clear cut... (considering only multi-word description to exclude simple package or component names which are always written lowercase)
```
buildroot ((2021.02))$ find . -name Config.in | xargs egrep '(bool|int|string).*\"[a-z].* .*' | wc -l
739
buildroot ((2021.02))$ find . -name Config.in | xargs egrep '(bool|int|string).*\"[A-Z].* .*' | wc -l
924
```
According to the same commands, the Linux kernel overwhelmingly uses capitalization (94.8 % of the time to be precise ;-)).

Comment (etienne-lms) [reply br-ext/package/optee_client_ext/Config.in:13]: fair

**Pull request merged**